packages/platform/atoms/calendar-settings/wrappers/CalendarSettingsPlatformWrapper.tsx

```diff
@@ -0,0 +1,240 @@
+                                    isChecked={cal.isSelected}
+                                    destination={cal.externalId === destinationCalendarId}
+                                    credentialId={cal.credentialId}
+                                  />
+                                );
+                              })}
+                            </ul>
+                          </div>
+                        </AppListCard>
+                      );
+                    }
+                    return (
+                      <Alert
+                        key={`alert-${connectedCalendar.credentialId}`}
+                        severity="warning"
+                        title={t("something_went_wrong")}
+                        message={<span>{t("calendar_error")}</span>}
+                        iconClassName="h-10 w-10 ml-2 mr-1 mt-0.5"
+                        actions={
+                          <div className="flex w-32 justify-end">
+                            <PlatformDisconnectIntegration
+                              credentialId={connectedCalendar.credentialId}
+                              trashIcon
+                              buttonProps={{ className: "border border-default" }}
+                              slug={connectedCalendar.integration.slug}
+                            />
+                          </div>
+                        }
+                      />
+                    );
+                  })}
+                </List>
+              </CalendarSettings>
+            );
+          }}
+        />
+      </div>
+    </AtomsWrapper>
+  );
+};
+
+const CalendarSettingsHeading = () => {
+  const { t } = useLocale();
+
+  return (
+    <div className="border-subtle border-b p-6">
+      <div className="flex items-center justify-between">
+        <div>
+          <h4 className="text-emphasis text-base font-semibold leading-5">{t("check_for_conflicts")}</h4>
+          <p className="text-default text-sm leading-tight">{t("select_calendars")}</p>
+        </div>
+      </div>
+    </div>
+  );
+};
+
+const PlatformDisconnectIntegration = (props: {
+  credentialId: number;
+  label?: string;
+  slug?: string;
+  trashIcon?: boolean;
+  isGlobal?: boolean;
+  onSuccess?: () => void;
+  buttonProps?: ButtonProps;
+}) => {
+  const { t } = useLocale();
+  const { onSuccess, credentialId, slug } = props;
+  const [modalOpen, setModalOpen] = useState(false);
+  const { toast } = useToast();
+  const { mutate: deleteCalendarCredentials } = useDeleteCalendarCredentials({
+    onSuccess: () => {
+      toast({
+        description: t("app_removed_successfully"),
+      });
+      setModalOpen(false);
+      onSuccess && onSuccess();
+    },
+    onError: () => {
+      toast({
+        description: t("error_removing_app"),
+      });
+      setModalOpen(false);
+    },
+  });
+
+  return (
+    <DisconnectIntegrationComponent
+      onDeletionConfirmation={async () => {
+        slug &&
```

Comment:
1. The disconnect button icons are missing
2. When i removed apple calendar it is still showing there but on bottom right "App removed successfully" was still showing.

<img width="1696" alt="Screenshot 2024-08-12 at 10 15 16" src="https://github.com/user-attachments/assets/cec367f9-54e3-4775-929c-59ac056a850e">

Reply:
1. Icons not appearing is a known issue that is caused by #16135, we also have a discussion in threads going on [here](https://app.campsite.co/cal/posts/0cvvend4juum) so that should be resolved soon
2. When you delete a calendar, it triggers the custom hook to delete the calendar credentials which takes a few seconds although the toast appears instantly. But since the web app follows the same behaviour thats why I implemented it the same way in the atom. The custom hook also has logic to invalidate connected calendar queries, so it refetches the connected calendars once the delete request is successful so technically the deleted calendar should be removed automatically.